### PR TITLE
Ghost roles (Developed by Drakos)

### PR DESCRIPTION
**Pull request by Tommy-XL**
## Thanks @Ultradragon005 for #537

Will be tested in 1.6.0 dev versions, if no problems are found it will be merged into main branch

### CODE REVIEW

File: Modules/GameOptionsSender/PlayerGameOptionsSender.cs [hunk 1]
```diff
@@ -100,21 +100,23 @@ public override IGameOptions BuildGameOptions()
                 opt.SetFloat(FloatOptionNames.ImpostorLightMod, 1.25f);
             }
         }
-        switch (role.GetCustomRoleTypes())
+        if (!role.IsGhostRole() || player.IsAnySubRole(x => x is CustomRoles.EvilSpirit))
```
Comment (Tommy-XL) [inline Modules/GameOptionsSender/PlayerGameOptionsSender.cs:103]: Incorrectly written conditions?
I mean, does it have to be like this?
`if (!(role.IsGhostRole() || player.IsAnySubRole(x => x is CustomRoles.EvilSpirit)))`

Comment (Tommy-XL) [reply Modules/GameOptionsSender/PlayerGameOptionsSender.cs:103]: Oh wain

Comment (Tommy-XL) [reply Modules/GameOptionsSender/PlayerGameOptionsSender.cs:103]: Forget about it lol

Comment (Ultradragon005) [reply Modules/GameOptionsSender/PlayerGameOptionsSender.cs:103]: ye `AURoleOptions.ShapeshifterCooldown = Options.DefaultShapeshiftCooldown.GetFloat();` was here so it checks for that

File: Patches/PlayerControlPatch.cs [hunk 2]
```diff
@@ -66,19 +88,16 @@ public static bool Prefix(PlayerControl __instance, [HarmonyArgument(0)] PlayerC
...
         {
-            if (__instance.Data.IsDead)
-            {
                 Logger.Info("Blocked protection", "CheckProtect");
-                return false;
-            }
+                return false; // What is this for? sheriff dosen't become guardian angel lmao
         }
+        
         return true;
```
Comment (Tommy-XL) [inline Patches/PlayerControlPatch.cs:101]: I think this code was needed before CrewmateGhost and ImpostorGhost appeared, so the sheriff was automatically assigned a Guardian Angel

Not sure about this, but I think so

**Comment by Tommy-XL** [comment]
Ok, I think it's ready for merget

**Pull request merged**